File: benchmarks/src/lua/gcd.lua
local start = os.clock()
local function gcd(a, b)
    while b ~= 0 do
        local temp = b
        b = a % b
        a = temp
    end
    return a
end
local result = 0
for i = 1, 10000 do
    result = gcd(i * 12345, i * 6789 + 1)
end
local finish = os.clock()
print("Result: " .. result)
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmarks/src/php/gcd.php
<?php
function gcd($a, $b) {
    while ($b != 0) {
        $t = $b; $b = $a % $b; $a = $t;
    }
    return $a;
}
$start = hrtime(true);
$result = 0;
for ($i = 1; $i <= 10000; $i++) {
    $result = gcd($i * 12345, $i * 6789 + 1);
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Result: " . $result . "
";
echo "Time: " . round($ms) . "ms
";
